mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java

```diff
@@ -743,12 +743,17 @@ public void testFindWithExplain() {
 			queryWithModifiers.append( ", '$explain': true " );
 			String nativeQuery = "db." + OscarWildePoem.TABLE_NAME + ".find({" + queryWithModifiers + "})";
 
-			NativeQuery query = session.createNativeQuery( nativeQuery );
-			@SuppressWarnings("unchecked")
-			List<Object[]> result = query.list();
-			// I'm not sure if we can test the content because this is the result of the explain command
-			// and I believe it might change among versions
-			assertThat( result.get( 0 ) ).isNotEmpty();
+			Object[] result = (Object[]) session.createNativeQuery( nativeQuery ).uniqueResult();
+			assertThat( result ).isNotEmpty();
+			// In MongoDB 7.0, the first entry of the result is "1" (I don't know why).
```

Comment:
I know why you're seeing this: some time after 3.6 the explain format totally changed, and the "first: (but see below) field is the explainVersion, which can either be 1 or 2 depending on configuration.  In my testing against a fresh 7.0 standalone, the value is actually 2 so the test fails.  I tried changing the logic to `result[0] instance of Number ? 1 : 0` to handle different explain versions, but then ran into a different problem: the order of the fields returned in `result` array does not appear to be stable. On some runs the value of `result[index]` is `Document`, as expected by the test, but in other runs it's `String`.  I'm not sure if this is due to to the server returning the fields in different order depending on some factor that we can't control, or whether it's Hibernate, but either way it doesn't seem safe to depend on the order.  

I suggest changing the test to something more flexible, perhaps like this:

```
			boolean foundExplainEvidence = false;
			for ( Object value : result ) {
				// Look for a document value containing the "namespace key". 
                                 // It's expected in the queryPlanner field of the explain output
				if ( value instanceof Document ) {
					Document possibleQueryPlannerDocument = (Document) value;
					if ( possibleQueryPlannerDocument.containsKey( "namespace" ) ) {
						foundExplainEvidence = true;
						break;
					}
				}
			}
			assertThat( foundExplainEvidence ).isEqualTo( true );
```

What do you think?

Reply:
Ok, thanks. Yes, I will update it.

Reply:
Should be fixed now: 21113763c0e007322d4a8d2ef0cfcf92b6028c79

If you could confirm that it works for you, I will merge this

Reply:
Confirmed.